packages/shared/lib/sdk/sync.ts

```diff
@@ -659,6 +648,23 @@ export class NangoAction {
         return this.nango.getToken(this.providerConfigKey, this.connectionId);
     }
 
+    /**
+     * Get current integration
+     */
+    public async getIntegration(queries?: GetPublicIntegration['Querystring']): Promise<GetPublicIntegration['Success']['data']> {
```

Comment:
This is just an aside: rather than sync providers.yaml it feels like we could just pull this data at runtime for each job and not have to load the whole providers file that way.

Reply:
yep agreed, there is a lot to improve regarding the task payload